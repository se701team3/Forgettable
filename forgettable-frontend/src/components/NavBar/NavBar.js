/* eslint-disable react/jsx-filename-extension */
/* eslint-disable max-len */
import React from 'react';
import Drawer from '@mui/material/Drawer';
import { List, ListItem } from '@mui/material';
import EncountersLogo from '../../assets/icons/navbar/encounters.svg';
import HomePageLogo from '../../assets/icons/navbar/homepage.svg';
import PersonsLogo from '../../assets/icons/navbar/persons.svg';
import SettingsLogo from '../../assets/icons/navbar/settings.svg';
import LightThemeLogo from '../../assets/logos/logo-short-black.svg';
import DarkThemeLogo from '../../assets/logos/logo-short-white.svg';
import classes from './NavBar.module.css';

const DRAWER_WIDTH = 128;
const ICON_SIZE = '26px';
const LOGO_SIZE = '50px';

export default function NavBar() {
  const [selectedIndex, setSelectedIndex] = React.useState(0); // @TODO: State stuff so that we can select the page
  const isLightTheme = true; // @TODO: When a state for dark theme is made, use this state so we can set the logo appropriately.
  const backgroundColor = getComputedStyle(document.documentElement).getPropertyValue('--bkgd');

  const linkProperties = [{
    src: HomePageLogo,
    alt: 'Home Page',
  }, {
    src: PersonsLogo,
    alt: 'Persons Page',
  }, {
    src: EncountersLogo,
    alt: 'Encounters Page',
  }, {
    src: SettingsLogo,
    alt: 'Settings Page',
  }];

  const handleListItemClick = (event, index) => {
    setSelectedIndex(index);
  };

  return (
    <Drawer
      sx={{
        width: DRAWER_WIDTH,
        flexShrink: 0,
        '& .MuiDrawer-paper': {
          width: DRAWER_WIDTH,
          boxSizing: 'border-box',
          backgroundColor,
        },
      }}
      variant="permanent"
      anchor="left"
    >
      <div className={classes.navBar_container}>
        <img
          src={isLightTheme ? LightThemeLogo : DarkThemeLogo}
          alt="Forgettable Logo"
          height={LOGO_SIZE}
          width={LOGO_SIZE}
          className={classes.navBar_logo}
        />
        <List>
          {linkProperties.map((linkItem, index) => (
            <ListItem
              button
              key={linkItem}
              onClick={(event) => handleListItemClick(event, index)}
              selected={selectedIndex === index}
              classes={{ selected: classes.navBar_listItem }}
            >
              <div className={classes.navBar_selectMarker} />
              <div className={classes.navBar_innerListItem}>
                <img
                  src={linkItem.src}
                  alt={linkItem.alt}
                  height={ICON_SIZE}
                  width={ICON_SIZE}
                  className={classes.navBar_linkIcon}
                />
              </div>
            </ListItem>
          ))}
        </List>
      </div>
    </Drawer>
  );
}
