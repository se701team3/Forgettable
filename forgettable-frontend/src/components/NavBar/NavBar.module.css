@import '../../colours.css';

.navBar_container {
    display: flex;
    flex-direction: column;
    gap: 30px;
}

.navBar_logo {
    padding: 70px 0px 23px 0px;
    margin: auto;
}

.navBar_innerListItem {
    width: 100%;
    text-align: center;
    padding: 15px 0px 15px 0px;
}

.navBar_linkIcon {
    filter: var(--ficon);
    position: relative;
    left: -5px; /* accounts for the selectMarker */
}

.navBar_selectMarker {
    width: 15px;
    height: 68px;
    background-color: var(--prmry);
    position: relative;
    left: -20px;
    border-radius: 0px 5px 0px 5px;
    visibility: hidden;
}

/* Selected item */
.navBar_listItem {
    background-color: var(--bkgd) !important;
}

.navBar_listItem .navBar_linkIcon {
    filter: var(--fprmry);
}

.navBar_listItem .navBar_selectMarker {
    visibility: visible;
}


