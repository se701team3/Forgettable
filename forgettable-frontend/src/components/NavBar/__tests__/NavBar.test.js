import ShallowRenderer from 'react-test-renderer/shallow';
import NavBar from '../NavBar';
import React from 'react';

// Need to mock the Link, as do not have the context of the Router
jest.mock('react-router-dom', () => ({
  Link: () => {
    const MockName = 'link-component';
    return <MockName />;
  },
}));

it('renders NavBar UI with correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<NavBar />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});


