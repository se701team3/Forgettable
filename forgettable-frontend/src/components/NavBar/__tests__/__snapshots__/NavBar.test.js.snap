// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBar UI with correct hierarchy 1`] = `
<ForwardRef(Drawer)
  anchor="left"
  sx={
    Object {
      "& .MuiDrawer-paper": Object {
        "backgroundColor": "",
        "boxSizing": "border-box",
        "width": 128,
      },
      "flexShrink": 0,
      "width": 128,
    }
  }
  variant="permanent"
>
  <div
    className="navBar_container"
  >
    <img
      alt="Forgettable Logo"
      className="navBar_logo"
      height="50px"
      src="logo-short-black.svg"
      width="50px"
    />
    <ForwardRef(List)>
      <Link
        to="/"
      >
        <ForwardRef(ListItem)
          button={true}
          classes={
            Object {
              "selected": "navBar_listItem",
            }
          }
          onClick={[Function]}
          selected={true}
        >
          <div
            className="navBar_selectMarker"
          />
          <div
            className="navBar_innerListItem"
          >
            <img
              alt="Home Page"
              className="navBar_linkIcon"
              height="26px"
              src="homepage.svg"
              width="26px"
            />
          </div>
        </ForwardRef(ListItem)>
      </Link>
      <Link
        to="/people"
      >
        <ForwardRef(ListItem)
          button={true}
          classes={
            Object {
              "selected": "navBar_listItem",
            }
          }
          onClick={[Function]}
          selected={false}
        >
          <div
            className="navBar_selectMarker"
          />
          <div
            className="navBar_innerListItem"
          >
            <img
              alt="People Page"
              className="navBar_linkIcon"
              height="26px"
              src="persons.svg"
              width="26px"
            />
          </div>
        </ForwardRef(ListItem)>
      </Link>
      <Link
        to="/encounters"
      >
        <ForwardRef(ListItem)
          button={true}
          classes={
            Object {
              "selected": "navBar_listItem",
            }
          }
          onClick={[Function]}
          selected={false}
        >
          <div
            className="navBar_selectMarker"
          />
          <div
            className="navBar_innerListItem"
          >
            <img
              alt="Encounters Page"
              className="navBar_linkIcon"
              height="26px"
              src="encounters.svg"
              width="26px"
            />
          </div>
        </ForwardRef(ListItem)>
      </Link>
      <Link
        to="/settings"
      >
        <ForwardRef(ListItem)
          button={true}
          classes={
            Object {
              "selected": "navBar_listItem",
            }
          }
          onClick={[Function]}
          selected={false}
        >
          <div
            className="navBar_selectMarker"
          />
          <div
            className="navBar_innerListItem"
          >
            <img
              alt="Settings Page"
              className="navBar_linkIcon"
              height="26px"
              src="settings.svg"
              width="26px"
            />
          </div>
        </ForwardRef(ListItem)>
      </Link>
    </ForwardRef(List)>
  </div>
</ForwardRef(Drawer)>
`;
